runtime/drivers/s3/s3.go

```diff
@@ -83,8 +124,8 @@ type Config struct {
 	url                   *globutil.URL
 }
 
-func ParseConfig(props map[string]any) (*Config, error) {
-	conf := &Config{}
+func parseConfig(props map[string]any) (*config, error) {
```

Comment:
Maybe rename to `parseSourceProperties` to disambiguate from the connector config?